src/main/java/com/minecolonies/entity/ai/citizen/builder/EntityAIStructureBuilder.java

```diff
@@ -820,14 +834,20 @@ private AIState completeBuild()
                                            job.getWorkOrderId()));
         }
 
-        job.complete();
         resetTask();
         worker.addExperience(XP_EACH_BUILDING);
         workFrom = null;
 
         return AIState.IDLE;
     }
 
+    private void cancelWork()
```

Comment:
I think we should just override resetTask, call super and add the rest of this. Seems like what this method was created for.